yarn-project/simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -60,6 +60,7 @@ abstract class ExternalCall extends Instruction {
     const l2Gas = memory.get(gasOffset).toNumber();
     const daGas = memory.getAs<Field>(gasOffset + 1).toNumber();
     const functionSelector = memory.getAs<Field>(this.temporaryFunctionSelectorOffset).toFr();
+    const isStaticCall = this.type === 'STATICCALL' || context.environment.isStaticCall ? 'STATICCALL' : 'CALL';
```

Comment:
Thanks for bringing this up! Let's discuss it on slack. (also please change the name XD and add a comment about the propagation).